Src/Base/AMReX_MFIter.H

```diff
@@ -180,6 +180,7 @@ protected:
     IndexType     typ;
 
     bool          dynamic;
+    bool          initialized;
```

Comment:
How about just
```suggestion
    bool          finalized = false;
```
Also note that this is not a static variable.  So we don't need to do anything special for OMP.  All we have to do is flip the flag in Finalize after testing the flag.